### PR TITLE
[Windows] Add Microsoft Reporting Services Projects support

### PR DESCRIPTION
**Pull request by AlexanderGitin**
# Description
[Windows] Add Microsoft Reporting Services Projects support

#### Related issue:
https://github.com/actions/virtual-environments/issues/5340
## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/toolsets/toolset-2022.json [hunk 1]
```diff
@@ -261,7 +261,8 @@
         ],
         "vsix": [
             "VisualStudioClient.MicrosoftVisualStudio2022InstallerProjects",
-            "WixToolset.WixToolsetVisualStudio2022Extension"
+            "WixToolset.WixToolsetVisualStudio2022Extension",
+            "ProBITools.MicrosoftReportProjectsforVisualStudio2022"
```
Comment (al-cheb) [inline images/win/toolsets/toolset-2022.json:265]: Please check - https://github.com/actions/virtual-environments/pull/5369

**Review by Nuttapon43 — APPROVED** [review]

**Review by azboyzranch — APPROVED** [review]

**Review by Nuttapon43 — APPROVED** [review]

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 2]
```diff
@@ -239,6 +239,12 @@ function Get-VsixExtenstionFromMarketplace {
     $request -match 'Microsoft\.VisualStudio\.Services\.Payload\.FileName":"(?<filename>[^"]*)' | Out-Null
     $fileName = $Matches.filename
     $downloadUri = $assetUri + "/" + $fileName
+    # ProBITools.MicrosoftReportProjectsforVisualStudio2022 has different URL https://github.com/actions/virtual-environments/issues/5340
+    if ($ExtensionMarketPlaceName -eq "ProBITools.MicrosoftReportProjectsforVisualStudio2022")
+    {
+        $fileName = "Microsoft.DataTools.ReportingServices.vsix"
+        $downloadUri = "https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ProBITools/vsextensions/MicrosoftReportProjectsforVisualStudio2022/3.0.1/vspackage"
```
Comment (al-cheb) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1:246]:
```
PS > curl.exe https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ProBITools/vsextensions/MicrosoftReportProjectsforVisualStudio2022/3.0.1/vspackage
{"$id":"1","innerException":null,"message":"Request was blocked due to exceeding usage of resource 'Count' in namespace 'AnonymousId'. For more information on why your request was blocked, see the topic \"Rate limits\" on the Microsoft Web site (https://go.microsoft.com/fwlink/?LinkId=823950).","typeName":"Microsoft.TeamFoundation.Framework.Server.RequestBlockedException, Microsoft.TeamFoundation.Framework.Server","typeKey":"RequestBlockedException","errorCode":0,"eventId":3000}
```
Some customers will face with  unable to download package.

Comment (miketimofeev) [reply images/win/scripts/ImageHelpers/InstallHelpers.ps1:246]: @AlexanderGitin we can use the direct link then
https://download.microsoft.com/download/b/b/5/bb57be7e-ae72-4fc0-b528-d0ec224997bd/Microsoft.DataTools.ReportingServices.vsix

**Review by miketimofeev — APPROVED** [review]

**Pull request merged**